### ISSUE TITLE
Early Stopping time - after 100,000 iterations 

### ISSUE THREAD

**Issue by awaisbajwaml** [tensorflow/tensorflow#55211]
**System information**
- I have used multiple custom classes to optimize the training time
- OS Linux Ubuntu 18.04):
- TensorFlow installed from binary
- TensorFlow 2.5
- Python 3.6:
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN 11.2 / 8.5
- GPU Test T4 

****

please see below 100k iterations, what is the time Early stopping is required to prevent overfitting. 

![image](https://user-images.githubusercontent.com/47017344/158029847-344d56f9-ad75-4d89-8deb-7708f6923f0f.png)

**Comment by Gelesh** [comment]
Is the plot, talking about training loss or validation loss. ?
Early stopping is required when, validation loss is increasing to answer 'what is the time Early stopping is required to prevent overfitting ? '.
Im sorry I dont understand the bug/issue being raised over here

**Comment by chunduriv** [comment]
@awaisbajwaml,

Generally we used `EarlyStopping` to stop training when a monitored metric has stopped improving. For more details you can refer [here](https://www.tensorflow.org/api_docs/python/tf/keras/callbacks/EarlyStopping).

Can you please elaborate your issue and share standalone code to investigate? Thanks!

**Comment by awaisbajwaml** [comment]
@Gelesh  this is training loss, not a validation one.

**Comment by Gelesh** [comment]
, To my understanding, Early stopping is supposed to configured to monitor the loss function on validation , and called from CallBacks.  Please refer my earlier comments.

" _Early stopping is required when, validation loss is increasing to answer 'what is the time Early stopping is required to prevent overfitting ? '. Im sorry I dont understand the bug/issue being raised over here_." 

@awaisbajwaml, Do you mean you have set up the EarlyStoppingCallBack , monitoring on Training Loss ? If so, I am not sure how would it serve the purpose of preventing over fitting. However, from the graph, the loss is decreasing, hence the trend shouldnt trigger any EarlyStop.

**Comment by chunduriv** [comment]
@awaisbajwaml,

Early stopping allows you to specify an arbitrarily large number of training epochs and stop training once the model performance stops improving on the validation dataset.

Follow the training procedure mentioned [here](https://www.tensorflow.org/tutorials/keras/overfit_and_underfit#training_procedure) to avoid overfitting with earlystopping function. Thanks!
